src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,330 @@
+            return;
+        }
+        if (seperatedWordsInContent[1].startsWith("by ")) {
+            String date = seperatedWordsInContent[1].split(" ", 2)[1];
+            if (date.trim().isEmpty()) {
+                throw new EmptyDescription();
+            }
+            String taskName = seperatedWordsInContent[0];
+            Task task = new Deadline(taskName,date);
+            tasks.add(task);
+            System.out.println(ADDING_TASK);
+            System.out.println("[D][] " + taskName + " (by: " + date + ")");
+            System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            String inputToDataFile = "D | " + convertMarkingStatusToNumber(task) + " | " + task.getContent()
+                    + " | " + task.getDate() + "\n";
+            appendTaskToDataFile(dataFile, inputToDataFile);
+        } else {
+            System.out.println(DEADLINE_INVALID_INPUT);
+        }
+    }
+
+    private static void addEventTask(String content, ArrayList<Task> tasks, FileClass dataFile)  {
+        String[] seperatedWordsInContent = content.split(" /");
+        if (doesIndexOutOfBoundsOccur(seperatedWordsInContent,2,EVENT_INVALID_INPUT)) {
+            return;
+        }
+        if (doesIndexOutOfBoundsOccur(seperatedWordsInContent[1].split(" ", 2),1,EVENT_INVALID_INPUT)) {
+            return;
+        }
+        String beginDate = seperatedWordsInContent[1].split(" ", 2)[1];
+        if (doesIndexOutOfBoundsOccur(seperatedWordsInContent[2].split(" ", 2),1,EVENT_INVALID_INPUT)) {
+            return;
+        }
+        String endDate = seperatedWordsInContent[2].split(" ", 2)[1];
+
+        if (seperatedWordsInContent[1].startsWith("from ") && seperatedWordsInContent[2].startsWith("to ")) {
+            String taskName = seperatedWordsInContent[0];
+            Task task = new Event(taskName,beginDate,endDate);
+            tasks.add(task);
+            System.out.println(ADDING_TASK);
+            System.out.println("[E][] " + taskName + " (from: " + beginDate + " to: " + endDate + ")");
+            System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            String inputToDataFile = "E | " + convertMarkingStatusToNumber(task) + " | " + task.getContent()
+                    + " | " + task.getBeginDate() + " | " + task.getEndDate() + "\n";
+            appendTaskToDataFile(dataFile, inputToDataFile);
+        } else {
+            System.out.println(EVENT_INVALID_INPUT);
+        }
+    }
+
+    private static void appendTaskToDataFile(FileClass dataFile, String inputToDataFile) {
+        try {
+            dataFile.appendToFile("src/main/data.txt", inputToDataFile);
+        } catch (IOException e) {
+            System.out.println(WRITEFILE_EXCEPTION_MESSAGE);
+        }
+    }
+
+    private static String convertMarkingStatusToNumber(Task task) {
+        if (task.getMarkingStatus().equals("[ ]")) {
+            return "0";
```

Comment:
Magic string here, consider making this into a named constant like how you did for other strings. Do look out for other magic strings/numbers/literals issues in this and other files.